nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,386 @@
+        <script
+        src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js">
+        </script>
+        <script src="https://cdn.plot.ly/plotly-gl3d-latest.min.js"></script>
+        <script>
+        {}
+        </script>
+        """.format(js_utils)
+    else:
+        with open(os.path.join(js_dir, 'jquery.min.js')) as f:
+            jquery = f.read()
+        with open(os.path.join(js_dir, 'plotly-gl3d-latest.min.js')) as f:
+            plotly = f.read()
+        js_lib = """
+        <script>{}</script>
+        <script>{}</script>
+        <script>
+        {}
+        </script>
+        """.format(jquery, plotly, js_utils)
+    return html.replace('INSERT_JS_LIBRARIES_HERE', js_lib)
+
+
+class SurfaceView(object):
+    """
+    Represents a web page.
+
+    use str(document) or document.html to get a web page,
+    document.get_iframe() to have it wrapped in an iframe.
+
+    """
+
+    def __init__(self, html, width=600, height=400):
+        self.html = html
+        self.width = width
+        self.height = height
+        self._temp_file = None
+
+    def resize(self, width, height):
+        self.width, self.height = width, height
+        return self
+
+    def get_iframe(self, width=None, height=None):
+        """
+        Get the document wrapped in an inline frame.
+
+        For inserting in another HTML page of for display in a Jupyter
+        notebook.
+
+        """
+        if width is None:
+            width = self.width
+        if height is None:
+            height = self.height
+        escaped = cgi.escape(self.html, quote=True)
+        wrapped = '<iframe srcdoc="{}" width={} height={}></iframe>'.format(
+            escaped, width, height)
+        return wrapped
+
+    def get_standalone(self):
+        """ Get the surface plot in an HTML page."""
+        return self.html
+
+    def _repr_html_(self):
+        return self.get_iframe()
+
+    def __str__(self):
+        return self.html
+
+    def save_as_html(self, file_name):
+        with open(file_name, 'wb') as f:
+            f.write(self.html.encode('utf-8'))
+
+    def open_in_browser(self, file_name=None):
+        if file_name is None:
+            fd, file_name = tempfile.mkstemp('.html', 'nilearn_surface_plot_')
+            os.close(fd)
+        self.save_as_html(file_name)
+        self._temp_file = file_name
+        file_size = os.path.getsize(file_name) / 1e6
+        print(("Saved HTML in temporary file: {}\n"
+               "file size is {:.1f}M, delete it when you're done!").format(
+                   file_name, file_size))
+        webbrowser.open(file_name)
+
+    def remove_temp_file(self):
+        if self._temp_file is None:
+            return
+        if not os.path.isfile(self._temp_file):
+            return
+        os.remove(self._temp_file)
+        print('removed {}'.format(self._temp_file))
+        self._temp_file = None
+
+
+def colorscale(cmap, values, threshold=None, symmetric_cmap=True):
```

Comment:
Same here about documentation. Minimal doc strings are super useful for future references.